Auto Grant Secret Read Access Docs (#179)
DataDog/datadog-cdk-constructs: README.md

## AWS CDK v1 vs AWS CDK v2
Two separate versions of Datadog CDK Constructs exist; `datadog-cdk-constructs` and `datadog-cdk-constructs-v2`. These are designed to work with `AWS CDK v1` and `AWS CDK v2` respectively.

> - `datadog-cdk-constructs-v2` requires Node 14+, while `datadog-cdk-constructs` supports Node 12+.

Reviewer: ```suggestion
- `datadog-cdk-constructs-v2` requires Node >= 14, while `datadog-cdk-constructs` supports Node >= 12.
```
We are trying to standardize the docs on using ">=" to describe version constraints going forward.

  site: "<SITE>",
  apiKey: "{Datadog_API_Key}",
  apiKeySecretArn: "{Secret_ARN_Datadog_API_Key}",

>   apiKeySecret: <AWS_CDK_ISECRET>, //Only available in datadog-cdk-constructs-v2

Reviewer: ```suggestion
  apiKeySecret: <AWS_CDK_ISECRET>, // Only available in datadog-cdk-constructs-v2
```

Add tags to your constructs. We recommend setting an `env` and `service` tag to tie Datadog telemetry together. For more information see [official AWS documentation][10] and [CDK documentation][11].

> ## Automatically Grant AWS Secret Read Access to Lambda Execution Role
> **Only Available in datadog-cdk-constructs-v2**

Reviewer: ```suggestion
## Automatically grant AWS secret read access to Lambda execution role
**Only available in datadog-cdk-constructs-v2**
```

> To automatically grant your lambda execution roles read access to a given secret, please pass in `apiKeySecret` in place of `apiKeySecretArn` when initializing the Datadog construct.

Reviewer: ```suggestion
To automatically grant your Lambda execution roles read access to a given secret, pass in `apiKeySecret` in place of `apiKeySecretArn` when initializing the Datadog construct.
```

```
const { Secret } = require('aws-cdk-lib/aws-secretsmanager');

const secret = Secret.fromSecretPartialArn(this, 'DatadogApiKeySecret' 'arn:aws:secretsmanager:us-west-1:123:secret:DATADOG_API_KEY');

const datadog = new Datadog(this, 'Datadog', {
  ...
  apiKeySecret: secret
  ...
});
```

> When `addLambdaFunctions` is called, the Datadog CDK construct will grant your lambda execution roles read access to the given AWS secret. This is done through the [AWS ISecret's grantRead function][17].

Reviewer: ```suggestion
When `addLambdaFunctions` is called, the Datadog CDK construct grants your Lambda execution roles read access to the given AWS secret. This is done through the [AWS ISecret's grantRead function][17].
```
